install/0000_30_machine-api-operator_02_machine.crd.yaml

```diff
@@ -7,23 +7,23 @@ metadata:
   name: machines.machine.openshift.io
 spec:
   additionalPrinterColumns:
-  - JSONPath: .status.providerStatus.instanceId
-    description: Instance ID of machine created in AWS
+  - JSONPath: .metadata.annotations['machine\.openshift\.io/instance-id']
```

Comment:
can we just get this from providerID?

Reply:
`ProviderID` is cloud provider prefixed, e.g. `aws://...`. My usual use case was to just copy paste the ID and use it to search instance in AWS portal. With `aws://` I would always have to remove the prefix. Also, for Azure it has no value as I don't think you can search Azure vm based on its ID. I tried and I failed. What would be the benefit of reading ProviderID?

Reply:
`providerID` is the machine API source of truth. It's build on top of the `instance-id` that you can just pick from it, hence no need nor value for an specific annotation

Reply:
I don't mind reading ProviderID instead. Though, the question is what's the user expectation of `InstanceID` field. Though, it does not make sense to name the column `Instance` anymore.

Also, as I asked in https://github.com/openshift/machine-api-operator/pull/360/files#r308229498, what would be the `providerID` value used for?

Reply:
We can put the providerID under `-o wide` [1] so it's not printed by default. Right now, it has value only for AWS.

[1] https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#priority

Reply:
FYI providerID is also in azure and must be in gcp.
ok cool I'm good with `instance-id'` please make sure to add that in azure https://github.com/openshift/cluster-api-provider-azure/commit/747894809218b1826aa4a9f454c25708c92e5052#diff-7c34c0cb525c96c7b732376b0c8994e1R56